tracer/src/Datadog.Trace/AppSec/SecuritySettings.cs

```diff
@@ -67,15 +72,40 @@ public SecuritySettings(IConfigurationSource? source, IConfigurationTelemetry te
                                             .WithKeys(ConfigurationKeys.AppSec.ObfuscationParameterValueRegex)
                                             .AsString(SecurityConstants.ObfuscationParameterValueRegexDefault, x => !string.IsNullOrWhiteSpace(x));
 
-            UserEventsAutomatedTracking = config
-                                         .WithKeys(ConfigurationKeys.AppSec.UserEventsAutomatedTracking)
-                                         .AsString(
-                                              UserTrackingSafeMode,
-                                              val =>
-                                                  val.Equals("disabled", StringComparison.OrdinalIgnoreCase)
-                                               || val.Equals(UserTrackingSafeMode, StringComparison.OrdinalIgnoreCase)
-                                               || val.Equals(UserTrackingExtendedMode, StringComparison.OrdinalIgnoreCase))
-                                         .ToLowerInvariant();
+            string GetLegacyUserTracking() =>
+                config
+                   .WithKeys(ConfigurationKeys.AppSec.UserEventsAutomatedTracking)
+                   .AsString(
+                        isPresent => isPresent ? UserTrackingDisabled : UserTrackingIdentMode,
+                        val =>
+                            val.Equals(UserTrackingDisabled, StringComparison.OrdinalIgnoreCase)
+                         || val.Equals(DeprecatedUserTrackingSafeMode, StringComparison.OrdinalIgnoreCase)
+                         || val.Equals(DeprecatedUserTrackingExtendedMode, StringComparison.OrdinalIgnoreCase))
+                   .ToLowerInvariant();
+
+            UserEventsAutoInstrumentationMode = config
+                                               .WithKeys(ConfigurationKeys.AppSec.UserEventsAutoInstrumentationMode)
+                                               .AsString(
+                                                    isPresent => isPresent ? UserTrackingDisabled : GetLegacyUserTracking(),
```

Comment:
This is somewhat of a fundamental change to the design of the `ConfigurationBuilder` and I'm not sure we should make it 😬 Also, it's kinda hard to follow, what behaviour are we trying to get here? I'm 90% sure we can do it without changing the builder...

Reply:
The config setting name was changed, if the new value is present we should use it, otherwise we should look for the old one.

The change allows us to do this all inside the configuration combinators, we could implement this logic without the changer but it would mean moving some of the defaults outside of the combinators so the changes would be visible to the telemetry.

Reply:
> The config setting name was changed, if the new value is present we should use it, otherwise we should look for the old one.

We have that capability already, by passing the key values to the `WithKeys()` and it's handled correctly by telemetry in that case 🤔 

e.g. see here, where we have two "deprecated" values that we fallback to: https://github.com/DataDog/dd-trace-dotnet/blob/0f912fefdd124405fe162c99458c9103ada53d82/tracer/src/Datadog.Trace/Configuration/TracerSettings.cs#L394

Reply:
As discussed on slack used `.AsStringResult` and rolled back the changes.